packages/hash/frontend/src/blocks/page/BlockContextMenu/BlockContextMenu.tsx

```diff
@@ -72,18 +75,22 @@ const BlockContextMenu: ForwardRefRenderFunction<
     const hasChildEntity =
       Object.keys(blockEntity?.properties.entity?.properties ?? {}).length > 0;
     const items = [
-      {
-        key: "set-entity",
-        title: hasChildEntity ? "Swap Entity" : "Add an entity",
-        icon: <FontAwesomeIcon icon={faAdd} />,
-        subMenu: (
-          <LoadEntityMenuContent
-            blockEntityId={entityId}
-            closeParentContextMenu={() => popupState.close()}
-          />
-        ),
-        subMenuWidth: 280,
-      },
+      ...(currentComponentId && !textBlockComponentIds.has(currentComponentId)
```

Comment:
`isHashTextBlock` exists

Reply:
https://github.com/hashintel/hash/pull/1014/commits/0f6b38a572f46b6190083c4edca2231134a1383f

(same commit for all changes btw)